### PR TITLE
[Core] Clear RuntimeOptions.featurePaths if rerun was used

### LINKED ISSUE #1630: All features are run instead of none
## Summary

If rerun report is empty and specified by `cucumber.options` property ALL features are run instead of none.
That doesn't happen when rerun file is specified in `@CucumberOptions` annotation.

## Expected Behavior

None of features are run.

## Current Behavior

All features are run when empty rerun report file is provided via `cucumber.options` property.

## Possible Solution

 - Parse `cucumber.options` property in `cucumber.runtime.RuntimeOptionsFactory` before running `addDefaultFeaturePathIfNoFeaturePathIsSpecified(args, clazz);`

## Steps to Reproduce

1. Create empty rerun file
2. Create testng cucumber test class
2. Set `cucumber.options` to `@empty-rerun-file.txt`
3. Run cucumber testng class

### PR DESCRIPTION
**Pull request by tommywo**
## Summary
Fixes: #1630

## How Has This Been Tested?
Unit tests and a test run on my laptop

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [x] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: core/src/main/java/cucumber/runtime/RuntimeOptions.java [hunk 1]
```diff
@@ -69,6 +69,7 @@ public String map(String keyword) {
     private boolean wip = false;
     private SnippetType snippetType = SnippetType.UNDERSCORE;
     private int threads = 1;
+    private boolean isRerun = false;
```
Comment (mpkorstanje) [inline core/src/main/java/cucumber/runtime/RuntimeOptions.java]: I think this may need to be method scoped.

**Comment by mpkorstanje** [comment]
Ah. Got a test case.

A cucumber annotation or CLI with a non-empty rerun file.

In combination with:

An environment variable without any features.

**Comment by tommywo** [comment]
I've added a test case for that and added one more - if the is rerun file passed via cli and a feature it should ignore rerun file from cli -similar like it does to features.

**Comment by mpkorstanje** [comment]
I'm having some trouble finding a test which uses:

CLI: `@file:path/rerun.txt`
ENVIRONMENT: `cucumber.options=--tags @smoke`

I think this will result in no feature files being selected to execute.

Could you point it out to me?

File: core/src/main/java/cucumber/runtime/RuntimeOptions.java [hunk 2]
```diff
@@ -67,6 +67,7 @@ public String map(String keyword) {
     private boolean strict = false;
     private boolean monochrome = false;
     private boolean wip = false;
+    private boolean wasRerun = false;
```
[the remark this answers is not in the document]
Comment (tommywo) [reply core/src/main/java/cucumber/runtime/RuntimeOptions.java]: right!

**Comment by tommywo** [comment]
There was no such test with rerun file so i've added one. In that case behaviour is the same like passing features with line to cli - line numbers are ignored and only tags are used.

**Comment by mpkorstanje** [comment]
All good now. Cheers!

**Pull request merged**